auth/src/main/java/org/aerogear/android/ags/auth/credentials/OIDCCredentials.java

```diff
@@ -0,0 +1,206 @@
+
+    /**
+     * OpenID Connect credentials containing the identity, refresh and access tokens provided on a
+     * successful authentication with OpenID Connect.
+     * @param serialisedCredential JSON string representation of the authState field produced by
+     *                             {@link #deserialize(String)}.
+     * @param integrityCheckParameters Integrity check parameters for the token.
+     */
+    public OIDCCredentials(final String serialisedCredential, final IIntegrityCheckParameters integrityCheckParameters) {
+        try {
+            this.authState = AuthState.jsonDeserialize(serialisedCredential);
+        } catch(JSONException e) {
+            throw new RuntimeException(e);
+        }
+        this.integrityCheckParameters = integrityCheckParameters;
+    }
+
+    public OIDCCredentials() {
+        this.authState = new AuthState();
+        this.integrityCheckParameters = new IntegrityCheckParameters();
+    }
+
+    public String getAccessToken() {
+        return authState.getAccessToken();
+    }
+
+    public String getIdentityToken() {
+        return authState.getIdToken();
+    }
+
+    public String getRefreshToken() {
+        return authState.getRefreshToken();
+    }
+
+    public IIntegrityCheckParameters getIntegrityCheckParameters() { return this.integrityCheckParameters; }
+
+    /**
+     * Verify the authenticity of a JWT token against integrity parameters (provided config).
+     * @param jwtToken The JWT token to verify.
+     * @return <code>true</code> if the token integrity is good.
+     */
+    public boolean verifyToken(final String jwtToken) {
+        final String issuer = integrityCheckParameters.getIssuer();
+        final String audience = integrityCheckParameters.getAudience();
+        final String publicKey = integrityCheckParameters.getPublicKey();
+        return verifyToken(jwtToken, publicKey, issuer, audience);
+    }
+
+    /**
+     * A function to verify the authenticity of a JWT token against a public key, expected issuer and audience.
+     * @param jwtToken - The JWT Token to Verify.
+     * @param publicKey - The Public Key from Keycloak, without the Begin/End tags.
+     * @param issuer - The expected Issuer of the JWT
+     * @param audience - The expected Audience of the JWT
+     * @return <code>true</code> if the token integrity is good.
+     */
+    public boolean verifyToken(final String jwtToken, final String publicKey, final String issuer, final String audience) {
+        final String constructedPublicKey = beginPublicKey + publicKey + endPublicKey;
+
+        // Convert the public key from a string to a Java security key
+        final RsaKeyUtil utils = new RsaKeyUtil();
+        final PublicKey jwtPublicKey;
+        try {
+            jwtPublicKey = utils.fromPemEncoded(constructedPublicKey);
+        } catch (JoseException e) {
+            Log.e(TAG, e.getMessage(), e);
+            throw new RuntimeException(e);
```

Comment:
Please use a more specific exception. In this case, if the parameters are not correct, an IllegalArgumentException could be the right one.